mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,415 @@
+method was presented which introduces **attention mechanism** to graphs, known as **[Graph attention networks (GAT)](https://arxiv.org/pdf/1710.10903.pdf?ref=https://githubhelp.com)**, by *Veličković
+et al*. The last two methods offer a great possibility for inductive learning. But they haven't been specifically developed
+to handle different events occurring on graphs, such as **node features updates**, **node deletion**, **edge deletion**
+and so on. These events happen regularly on **real-world** examples such as **[Twitter network](https://twitter.com/memgraphmage)**, 
+where users update their profile, delete their profile or just unfollow another user.
+
+In their work, Rossi et al introduced to us [Temporal graph networks](https://arxiv.org/abs/2006.10637) which presents a great possibility to do graph machine
+learning on a stream of data, use-case occurring more often in recent years.
+
+### About query module
+What we have covered in this module
+  * **link prediction** - train your **TGN** to predict new **links/edges** and **node classification** - predict labels of nodes from graph structure and **node/edge** features
+  * **graph attention layer** embedding calculation and **graph sum layer** embedding layer calculation
+  * **mean** and **last** as message aggregators
+  * **mlp** and **identity(concatenation)** as message functions
+  * **gru** and **rnn** as memory updater
+  * **uniform** temporal neighborhood sampler
+  * **memory** store and **raw message store**
+
+as introduced by **[Rossi et al](https://emanuelerossi.co.uk/)**.
+
+Following means **you** can use **TGN** to be able to **predict edges** or  to perform **node classification** tasks, with **graph attention layer** or **graph sum layer**, by using
+either **mean** or **last** as message aggregator, **mlp** or **identity** as message function, and finally  **gru** or **rnn** as memory updater.
+
+In total that gives *you* **2\*2\*2\*2\*2 options**, in total of **32** options to explore on your graph :smile: 
+
+To start exploring our module, **[github/memgraph/mage](https://github.com/memgraph/mage)** and start exploring our implementation, or even better, jump to 
+**[download page](https://memgraph.com/download)**, Download **Memgraph platform** and start exploring **TGN**
+
+
+What is **not** implemented in the module:
+  * **node update/deletion events** since they occur very rarely - although we have prepared a terrain
+  * **edge deletion** events 
+  * **time projection** embedding calculation and **identity** embedding calculation since author mentions 
+    they perform very poorly on all datasets - although it is trivial to add new layer
+
+
+What we **believe** we offer instead of authors implementation from **[twitter-research GitHub repo](https://github.com/twitter-research/tgn)**:
```

Comment:
I would remove this paragraph, when working on a solution for developers we need to be certain and not believ, which we are currently not.

Reply:
I am 99% sure they have a bug, but as you can't be 100% sure, I will remove it 👍🏻